lua/apisix/plugins/basic-auth.lua

```diff
@@ -0,0 +1,225 @@
+local core = require("apisix.core")
+local ngx = ngx
+local ngx_re = require("ngx.re")
+local json = require("apisix.core.json")
+
+local authorizations_etcd
+
+local lrucache = core.lrucache.new({
+    ttl = 300, count = 512
+})
+
+-- You can follow this document to write schema:
+-- https://github.com/Tencent/rapidjson/blob/master/bin/draft-04/schema
+-- rapidjson not supported `format` in draft-04 yet
+local schema = {
+    type = "object",
+    properties = {
+        enable = { type = "boolean", default = true, enum = { true, false } },
```

Comment:
I think the properties should be
```
username = {type = "string"},
password = {type = "string"}
```

otherwise how to set user and password using admin API?